ext/date/tests/gh14383.phpt

```diff
@@ -0,0 +1,39 @@
+--TEST--
+Bug GH-14383 (DateTime::createFromTimestamp overflowed microseconds value)
+--INI--
+date.timezone=UTC
+--FILE--
+<?php
+$cases = [
+    [0.999_999_0, '0.999_999_0'],
+    [0.999_999_1, '0.999_999_1'],
+    [0.999_999_8, '0.999_999_8'],
+    [0.999_999_9, '0.999_999_9'],
+    [1.000_000_0, '1.000_000_0'],
+    [1.000_000_1, '1.000_000_1'],
+    [1.000_000_8, '1.000_000_8'],
+    [1.000_000_9, '1.000_000_9'],
+    [1.000_001_0, '1.000_001_0'],
+    [1.000_001_1, '1.000_001_0'],
+    [1.000_001_8, '1.000_001_8'],
+    [1.000_001_9, '1.000_001_9'],
+];
+
+foreach ($cases as [$usec, $label]) {
+    echo "{$label}: ";
+    echo DateTime::createFromTimestamp($usec)->format('s.u'), "\n";
+}
```

Comment:
```suggestion

foreach ($cases as [$usec, $label]) {
    echo "{$label}: ";
    echo DateTime::createFromTimestamp($usec)->format('s.u'), "\n";
}

foreach ($cases as [$usec, $label]) {
    echo "-{$label}: ";
    echo DateTime::createFromTimestamp(-$usec)->format('s.u'), "\n";
}
```

Reply:
Ah, right. I didn't consider the negative case.

Reply:
However, we should probably change the value of sec a little in the negative case. Underflows can occur, making it difficult to understand the expected value.

Reply:
After all, it was inevitable that it would be a bit hard to see...